### ISSUE TITLE
PWA: Interim modal for User Profile email address warning messages and Re- deploy Sendgrid Validation 

### ISSUE THREAD

**Issue by rohanjay10** [episphere/connect#1181]
(Ops suggestion is to put this on the backlog for a later release)

Interim modal that would appear after User Profile is submitted, but before the UP review page, to indicate emails deemed risky by Sendgrid Email Validation. 

- "This email address may be invalid. Please double check your entry before continuing"
Spanish: "Esta dirección de correo electrónico puede no ser válida. Por favor, vuelva a revisar su entrada antes de continuar."

![Image](https://github.com/user-attachments/assets/20394e2e-bc26-4f92-8ec8-b4657173e626)

**Comment by depietrodeanna** [comment]
@sonyekere I do not see a need to update this modal language. I can get this translated to Spanish and add that here. @robertsamm do you agree the language is OK?

**Comment by robertsamm** [comment]
@depietrodeanna I agree the language in the message looks ok, should we change the all caps 'WARNING' at the top of the modal?

**Comment by depietrodeanna** [comment]
yes definitely- I don't think we need a title necessarily. Dev team, can we simply remove "Warning" from this modal?

**Comment by rohanjay10** [comment]
Hi @jhflorey could we remove the "WARNING" portion at the top out of the interim warning modal here (part of the February release)

![Image](https://github.com/user-attachments/assets/b7440b7e-ab01-40ae-85ac-d6950fd5ae9a)

**Comment by HanaShiho** [comment]
Thanks @rohanjay10 I'm fine with removing the title from the modal

**Comment by rohanjay10** [comment]
No problem @HanaShiho . Hi @jhflorey I think we can remove the "WARNING" from the warning modal box with the consensus

> Hi [@jhflorey](https://github.com/jhflorey) [@HanaShiho](https://github.com/HanaShiho) could we remove the "WARNING" portion at the top out of the interim warning modal here (part of the February release)
> 
> ![Image](https://github.com/user-attachments/assets/b7440b7e-ab01-40ae-85ac-d6950fd5ae9a)

**Comment by jhflorey** [comment]
@rohanjay10 Got it.

**Comment by jhflorey** [comment]
@rohanjay10  @robertsamm  @depietrodeanna Does it look good?

<img width="697" alt="Image" src="https://github.com/user-attachments/assets/8409b71a-c71b-49c4-9410-5e4d0358b1e4" />

<img width="671" alt="Image" src="https://github.com/user-attachments/assets/1d0e627f-44ac-4389-af1d-b9312e1a7b42" />

**Comment by depietrodeanna** [comment]
@jhflorey yes looks good to me,

**Comment by robertsamm** [comment]
@jhflorey agreed - looks good!

**Comment by jhflorey** [comment]
@robertsamm @depietrodeanna @rohanjay10 my code changes are ready in dev. We can test it.

**Comment by HanaShiho** [comment]
Thanks, @jhflorey. The original post says it will appear when email address are deemed risky; so does that mean email addresses tagged as risky by SendGrid will get the warning message, while invalid emails get nothing? Also, would this only apply to new account creation, or would it also appear when the preferred email address is updated in the existing UP?

**Comment by jhflorey** [comment]
@HanaShiho the invalid email will have the red error text below this email textbox.
@rohanjay10 Do we need to implement this feature for updating email in my profile page?

**Comment by rohanjay10** [comment]
Hi @HanaShiho this is the error text Jessica mentioned above:

<img width="556" alt="Image" src="https://github.com/user-attachments/assets/aa26b8ea-f1cf-4130-bb54-fd249aa03d38" />
@jhflorey for your question we'd need input from @robertsamm @HanaShiho @depietrodeanna if the warning message/modal needs to be implemented in MyProfiles page when pts are editing their email. Invalid emails are shown the error message above.

@sonyekere mentioned that the Interim warning modal would be part of the March release, so if it also needed to be implemented in MyProfiles I assume it would be a lower priority after it was implemented in the main User Profile page.
Please let me know if the above info needs to be updated.

**Comment by HanaShiho** [comment]
Thanks, Rohan. @brotzmanmj also confirmed we currently have a warning for invalid messages in prod, so the above change will be tested with risky messages. 

My question is, if someone updated their preferred email via SMDB or PWA, would that updated email address be validated by SendGrid? I believe when the existing preferred email is updated in PWA, it must pass SG validation, but @jhflorey @rohanjay10, can you please confirm for both SMDB and PWA? This will help clarify my last question about how a warning msg should behave when updating the existing UP.

**Comment by sonyekere** [comment]
Hello All. @jhflorey , @rohanjay10  and @HanaShiho a decision has been made to pause the conversation on this. There's not enough time or bandwidth to address the follow-up questions before the dev freeze on 2/14. This will be addressed in March. Thank you!

**Comment by jhflorey** [comment]
@HanaShiho just want to confirm

-  In the SMDB we have not implement email validation.
-  In the PWA user cannot update the existing preferred email if it is invalid.

**Comment by jhflorey** [comment]
@rohanjay10 do we need to implement email validation for SMDB

**Comment by sonyekere** [comment]
In addition to the User Profile and MyProfile modal request, a modal should be added to the user interface for fields that had email validation, including the 'Alt email' field. Rohan and Sandra will connect on Monday to finalize the requirements.

**Comment by brotzmanmj** [comment]
To clarify, it is the same modal that is needed for 'risky' emails as returned by the validation for all emails in the User Profile and MyProfile that have SendGrid email validation on them. It's not a different modal. Let me know if any questions.

**Comment by brotzmanmj** [comment]
@rohanjay10  We do not believe we have SendGrid email validation on the 'alternate contacts' email fields, and therefore there would not be a need for the warning modal. Can you confirm?

**Comment by jhflorey** [comment]
@rohanjay10 The warning modal for MyProfiles is ready in dev. Please check it.

**Comment by robertsamm** [comment]
@robertsamm to assist with testing for this from Ops.

**Comment by HanaShiho** [comment]
The SG validation tool is down. I tried testing Jessica's fix for warning messages for risky emails and noticed the warning message didn't appear. I opened a support ticket.

**Comment by hullingsag** [comment]
Hi all, apologies for the incorrect status update - I am syncing issues from the Connect & Development project board onto the Analytics Team Issue Tracker board. It looks like Github automatically assigns the status in a board to best match changes in the other, which I think is what happened here. I've now moved this issue back to "In Progress" in the Analytics Team Issue Tracker board so that it does not appear as completed. Thank you!

**Comment by sonyekere** [comment]
Testing for the March release will be paused and revised in April

**Comment by jhflorey** [comment]
my change is ready for stage

**Comment by jhflorey** [comment]
Currently we have issue with sendgrid email authentication API (403 error) so i have commented out the code that uses sendgrid API. Once we fix 403 error i will uncomment my code to make it work again.

**Comment by jhflorey** [comment]
The 403 issue has been fixed. i will update code to turn on the SG email validation

**Comment by sonyekere** [comment]
Development has been completed, pending PR.

**Comment by jhflorey** [comment]
@rohanjay10 the interim modal is ready in dev.

**Comment by robertsamm** [comment]
@jhflorey @rohanjay10 I have tested the interim modal for the soft error message and this looks good in dev. Did anything need to be retested for the re-deploy to sendgrid validation or is that in a separate github issue?

**Comment by jhflorey** [comment]
Thx @robertsamm for testing. Rohan will complete his testing for extra case next week. 
Having nice weekend team.

**Comment by robertsamm** [comment]
I have tested the interim modal for the soft error message and this looks good in stage. @rohanjay10 will be doing the rest of the testing for the SG validation in stage.
